code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			user << "<span class='warning'>You need to dry this first!</span>"
 	else
 		..()
+
+///////////////
+//VAPE NATION//
+///////////////
+/obj/item/clothing/mask/cigarette/vape
+	name = "E-Cigarette"
+	desc = "A classy and highly sophisticated electronic cigarette, for classy and dignified gentlemen. A warning label reads \"Warning: do not fill with flamable materials\""//<<< i'd vape to that.
+	icon_state = null
+	item_state = null
+	icon_on = null  //Note - these are in masks.dmi
+	icon_off = null // i'm lazy
+	smoketime = 3000 //long live the vape
+	chem_volume = 100
+	var/vapetime = 0 //this so it won't puff out clouds every tick
+	var/screw = 0 // kinky
+	var/super = 0 //for the fattest vapes dude.
+	var/emagged = 0 //LET THE GRIEF BEGIN
+
+/obj/item/clothing/mask/cigarette/vape/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is puffin hard on dat vape, they trying to join the vape life on a whole notha plane!")//it doesn't give you cancer, it is cancer
+	return (TOXLOSS|OXYLOSS)
+
+
+/obj/item/clothing/mask/cigarette/vape/New(loc, var/param_color = null)
+	..()
+	reagents.add_reagent("nicotine", 50)
+	if(!icon_state)
+		if(!param_color)
+			param_color = pick("red","blue","black","white","green","purple","yellow","orange")
+		icon_state = "[param_color]_vape"
+		icon_on = "[param_color]_vape"
+		icon_off = "[param_color]_vape" //i'm extremely lazy
+		item_state = "[param_color]_vape"
+
+/obj/item/clothing/mask/cigarette/vape/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/reagent_containers))
+		if(reagents.total_volume < chem_volume)
+			if(O.reagents.total_volume > 0)
+				O.reagents.trans_to(src,25)
+				user << "<span class='notice'>You add the contents of [O] to the [src]</span>"
+			else
+				user << "<span class='warning'>The [O] is empty!</span>"
+		else
+			user << "<span class='warning'>[src] can't hold anymore reagents!</span>"
+
+	if(istype(O, /obj/item/weapon/screwdriver))
+		if(!screw)
+			screw = 1
+			user << "<span class='notice'>You open the cap on the [src]</span>"
+			if(super)
+				add_overlay(image(icon, "vapeopen_med"))
+			else
+				add_overlay(image(icon, "vapeopen_low"))
+		else
+			screw = 0
+			user << "<span class='notice'>You close the cap on the [src]</span>"
+			cut_overlays()
+
+	if(istype(O, /obj/item/device/multitool))
+		if(screw && !emagged)//also kinky
+			if(!super)
+				cut_overlays()
+				super = 1
+				user << "<span class='notice'>You increase the voltage in the [src]</span>"
+				add_overlay(image(icon, "vapeopen_med"))
+			else
+				if(!emagged)

```

Comment:
The else is there to keep it from being lowered if it's already been emagged. Basically if you emag it, it's broken.